### PR TITLE
kernel: checkout patched based on kata_version

### LINKED ISSUE #857: release workflow fails due to unbound variable in kernel build
# Description of problem

I tagged the kata repositories and tagged my fork of kata-containers/kata-containers to run a "nightly" test of all components with kata-deploy (ie, our release workflow with the actual pushing to DockerHub and GitHub removed).

I noticed that a [change](https://github.com/kata-containers/packaging/commit/ff20f2008b8d53ff49f3d115d1ae71ce061d7bc6) in the build scripts introduces an unbound variable, causing the release process to fail.  We need this fixed to facilitate next release.

# Expected result

Green check marks

# Actual result

Red X.  From https://github.com/egernst/kata-containers/runs/338535212:
```
~/go/src/github.com/kata-containers/packaging ~/go/src/github.com/kata-containers/packaging/release ~/work/kata-containers/kata-containers
INFO: build kernel
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   604    0   604    0     0   2530      0 --:--:-- --:--:-- --:--:--  2537

100 5512k  100 5512k    0     0  11.7M      0 --:--:-- --:--:-- --:--:-- 11.7M
versions file (versions-master.yaml) does not exist
Download from https://raw.githubusercontent.com/kata-containers/runtime/master/versions.yaml
INFO: Clone config and patches
./kernel/build-kernel.sh: line 275: NEW_VERSION: unbound variable
```

### PR DESCRIPTION
**Pull request by amshinde**
NEW_VERSION may be unbound whereas kata_version should be defined
following manual release process docs and while using github actions.
Use kata_version instead to checkout correct version of patches.
Check if kata_version is not empty before doing so,
as the release may be triggered for master as well.

Fixes #857

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by amshinde** [comment]
I'll backport once this is merged.

File: kernel/build-kernel.sh [hunk 1]
```diff
@@ -272,10 +272,12 @@ get_config_and_patches() {
 		info "Clone config and patches"
 		patches_path="${default_patches_dir}"
 		if [ ! -d "${patches_path}" ]; then
-			tag="${kata_version:-$NEW_VERSION}"
+			tag="${kata_version}"
 			git clone "https://${patches_repo}.git" "${patches_repo_dir}"
 			pushd "${patches_repo_dir}" >> /dev/null
-			git checkout $tag
+			if [ ! -z $tag ] ; then
```
Comment (chavafg) [inline kernel/build-kernel.sh]: this could be `-n`, but fine if want to merge this way.

Comment (amshinde) [reply kernel/build-kernel.sh]: fixed

**Comment by chavafg** [comment]
I'm seeing this error:

```
15:26:15 Check tag_repos.sh create tags but not push
15:26:17 Already on 'master'
15:26:17 error: gpg failed to sign the data
15:26:17 error: unable to sign the tag
15:26:17 /tmp/jenkins/workspace/kata-containers-packaging-ubuntu-18-04-PR/go/src/github.com/kata-containers/packaging/.ci/../Makefile:27: recipe for target 'test-release-tools' failed
```

**Comment by jcvenegas** [comment]
OBS failed, looks that is also related last release is not tagged, 
https://github.com/kata-containers/packaging/pull/862 see for more debug

**Pull request merged**